src/Microsoft.AspNetCore.Server.Kestrel/Filter/Internal/LibuvStream.cs

```diff
@@ -132,5 +132,83 @@ private ValueTask<int> ReadAsync(ArraySegment<byte> buffer)
         {
             return _input.ReadAsync(buffer.Array, buffer.Offset, buffer.Count);
```

Comment:
`ChannelStream`? I could also move `SocketInput` out of the `Http` folder, but didn't feel it was related to this change. Maybe it's worth moving everything around and cleaning up as part of this PR.